rust/polars-ocaml/src/sql_context.rs

```diff
@@ -43,6 +43,34 @@ fn rust_sql_context_register(
     OCaml::unit()
 }
 
+#[ocaml_interop_export]
+fn rust_sql_context_execute_with_data_frames(
+    cr: &mut &mut OCamlRuntime,
+    data_frames: OCamlRef<OCamlList<DynBox<crate::data_frame::PolarsDataFrame>>>,
+    names: OCamlRef<OCamlList<String>>,
+    query: OCamlRef<String>,
+) -> OCaml<Result<DynBox<crate::data_frame::PolarsDataFrame>, String>> {
+    let data_frames: Vec<crate::data_frame::PolarsDataFrame> = 
+        unwrap_abstract_vec(data_frames.to_rust(cr));
+    let names: Vec<String> = names.to_rust(cr);
+    let query: String = query.to_rust(cr);
+
+    let mut sql_context = SQLContext::new();
+    let _ = data_frames
+        .into_iter()
+        .zip(names.iter())
```

Comment:
Instead of unzipping on the OCaml side and re-zipping on the Rust side, could we pass a `OCamlRef<OCamlList<(String, DynBox<crate::data_frame::PolarsDataFrame>)>>`? I guess it's slightly more involved since we won't be able to use `unwrap_abstract_vec` but I think it won't be too painful (and I think the result will be more cleaner).

Separately, since we don't care about the eventual result let's either use a `for` loop (preferred) or `.for_each()`.

Reply:
I made some partial progress here but I ran into the following error:

the size for values of type `str` cannot be known at compilation time
note: only the last element of a tuple may have a dynamically sized type

It seems that "name" is a string, which doesn't seem to have a size known at compile-time? Is there a rust alternative to tuple that works better for our use case?

Reply:
I think this is working now?